intelmq/bots/outputs/cif3/output.py

```diff
@@ -0,0 +1,237 @@
+
+SPDX-License-Identifier: AGPL-3.0-or-later
+SPDX-FileCopyrightText: 2022 REN-ISAC
+
+A shortened copy of this documentation is kept at `docs/user/bots.rst`, please
+keep it current, when changing something.
+
+Parameters:
+  - add_feed_provider_as_tag: bool, use false when in doubt
+  - cif3_additional_tags: list of tags to set on submitted indicator(s)
+  - cif3_feed_confidence: float, used when mapping a feed's confidence fails or
+        if static confidence param is true
+  - cif3_static_confidence: bool (use false when in doubt)
+  - cif3_token: str, API key for accessing CIF
+  - cif3_url: str, URL of the CIFv3 instance
+  - fireball: int, used to batch events before submitting to a CIFv3 instance
+        (default is 500 per batch, use 0 to disable batch and send each event as received)
+  - http_verify_cert: bool, used to tell whether the CIFv3 instance cert should be verified
+        (default true, but can be set to false if using a local test instance)
+
+Example (of some parameters in JSON):
+
+    "add_feed_provider_as_tag": true,
+    "cif3_additional_tags": ["intelmq"]
+
+
+"""
+try:
+    import ujson as json
+except ImportError:
+    JsonLib = None
+
+from datetime import datetime
+from typing import Optional
+
+from intelmq.lib.bot import OutputBot
+from intelmq.lib.exceptions import IntelMQException, MissingDependencyError
+
+try:
+    from cifsdk.client.http import HTTP as HttpClient
+    from csirtg_indicator import Indicator
+    from cifsdk._version import get_versions as get_cifsdk_version
+except ImportError:
+    HttpClient = None
+
+INTELMQ_TO_CIF_FIELDS_MAP = {
+    'source.ip': 'indicator',
+    'source.fqdn': 'indicator',
+    'source.url': 'indicator',
+    'source.network': 'indicator',
+    'source.port': 'port',
+    'feed.url': 'reference',
+    'time.source': 'lasttime',
+    'time.observation': 'reporttime',
+    'event_description.text': 'description',
+    'event_description.url': 'reference',
+    'malware.hash.md5': 'indicator',
+    'malware.hash.sha1': 'indicator',
+    'malware.hash.sha256': 'indicator',
+    'malware.name': 'description',
+    'protocol.application': 'application',
+    'protocol.transport': 'protocol',
+    'tlp': 'tlp',
+}
+
+INTELMQ_CLASSIFICATION_TO_CIF_TAGS_MAP = {
+    'c2-server': 'botnet,c2',
+    'undetermined': 'suspicious',
+    'brute-force': 'bruteforce',
+    'other': 'suspicious',
+}
+
+
+class CIF3OutputBot(OutputBot):
+    """
+    Submits indicators to a CIFv3 instance
+
+
+    IntelMQ-Bot-Name: CIFv3 API
+    """
+    add_feed_provider_as_tag: bool = False
+    cif3_feed_confidence: float = 5
+    cif3_static_confidence: bool = False
+    cif3_additional_tags = []
+    cif3_token: str = None
+    cif3_url: Optional[str] = None
+    fireball: int = 500
+    http_verify_cert: bool = True
+
+    _is_multithreadable = False
+
+    def init(self):
+        cifsdk_version = int(get_cifsdk_version().get('version').split('.')[0])
```

Comment:
If `cifsdk` is not installed, this will fail. Put the `if HttpClient is None` line above this one
```
Traceback (most recent call last):
  File "/.../intelmq/intelmq/lib/bot.py", line 214, in __init__
    self.init()
  File "/.../intelmq/intelmq/bots/outputs/cif3/output.py", line 94, in init
    cifsdk_version = int(get_cifsdk_version().get('version').split('.')[0])
NameError: name 'get_cifsdk_version' is not defined
```

Reply:
addressed in 9a0c33cca34c54fb7f86a8b35b99d9e721d3a257